src/CortexMock.cpp

```diff
@@ -0,0 +1,855 @@
+  // extractBodyDefs(body_defs_, document_["bodyDefs"]);
+}
+
+CortexMock::CortexMock(const CortexMock & other)
+: verbosity_level_(other.verbosity_level_), analog_bit_depth_(other.analog_bit_depth_),
+  conv_rate_to_mm_(other.conv_rate_to_mm_), frame_rate_(other.frame_rate_), analog_sample_rate_(
+    other.analog_sample_rate_),
+  axis_up_(other.axis_up_), dataHandlerFunc_(other.dataHandlerFunc_), errorMsgHandlerFunc_(
+    other.errorMsgHandlerFunc_), capture_file_name_(other.capture_file_name_),
+  current_framenum_(other.current_framenum_)
+{
+  // the result of the copy-constructor isn't active yet, even if the other one was active
+  // TODO(Gergely Kovacs) if we connect to client, addresses and ports need to be copied, too
+  initReadFile();
+}
+
+void CortexMock::swap(CortexMock & other) const noexcept
+{
+  // E.g. if pointers need to be swapped
+}
+
+CortexMock & CortexMock::operator=(CortexMock other)
+{
+  other.swap(*this);  // Copy-constructor and non-throwing swap
+  return *this;
+}
+
+CortexMock::~CortexMock()
+{
+  run_thread.join();
+  freeFrame(&current_frame_);
+}
+
+void CortexMock::getCaptureFilename(std::string & dest) const
+{
+  dest = capture_file_name_;
+}
+
+int CortexMock::getSdkVersion(unsigned char version[4]) const
+{
+  version[0] = 0;
+  version[1] = 8;
+  version[2] = 0;
+  version[3] = 0;
+  return RC_Okay;
+}
+
+int CortexMock::setVerbosityLevel(int i_level)
+{
+  verbosity_level_ = i_level;
+  return RC_Okay;
+}
+
+int CortexMock::getVerbosityLevel() const
+{
+  return verbosity_level_;
+}
+
+void CortexMock::errorMsgInString(int i_level, std::string & msg_str) const
+{
+  if (verbosity_level_ >= i_level) {
+    errorMsgHandlerFunc_(i_level, &msg_str[0]);
+  }
+}
+
+int CortexMock::setMinTimeout(int ms_timeout)
+{
+  // Timeout has no use here actually,
+  // because we don't communicate with the host
+  std::string error_msg = "No use of timeout in mock version";
+  errorMsgInString(VL_Error, error_msg);
+  return RC_ApiError;
+}
+
+int CortexMock::getMinTimeout() const
+{
+  // Timeout has no use here actually,
+  // because we don't communicate with the host
+  std::string error_msg = "No use of timeout in mock version";
+  errorMsgInString(VL_Error, error_msg);
+  return RC_ApiError;
+}
+
+int CortexMock::setErrorMsgHandlerFunc(
+  void (* errorMsgHandlerFunc)(int i_log_level,
+  char * sz_log_message))
+{
+  errorMsgHandlerFunc_ = errorMsgHandlerFunc;
+  return RC_Okay;
+}
+
+int CortexMock::setDataHandlerFunc(void (* dataHandlerFunc)(sFrameOfData * p_frame_of_data))
+{
+  dataHandlerFunc_ = dataHandlerFunc;
+  return RC_Okay;
+}
+
+int CortexMock::sendDataToClients(sFrameOfData * p_frame_of_data) const
+{
+  // TODO(Gergely Kovacs) send through TCP if client communication is going to be enabled
+  std::string error_msg = "No communication with client in mock version";
+  errorMsgInString(VL_Error, error_msg);
+  return RC_ApiError;
+}
+
+void CortexMock::setClientCommunicationEnabled(int b_enabled)

```

Comment:
Very good question, I don't know why it is implemented this way in the SDK. If it was consistent and logical, this along with setThreadPriorities should have int as return value instead of void, just like all of the other functions.